src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,150 @@
-/* eslint-disable */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  fetchTodosStart,
+  fetchTodosSuccess,
+  fetchTodosFailure,
+} from '../../features/todos';
+import { getTodos, getUser } from '../../api';
+import { RootState } from '../../app/store';
+import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { TodoModal } from '../TodoModal';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useDispatch();
+  const { todos, loading, error } = useSelector(
+    (state: RootState) => state.todos,
+  );
+  const { status, query } = useSelector((state: RootState) => state.filter);
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      dispatch(fetchTodosStart());
+      try {
+        const todosData = await getTodos();
+
+        dispatch(fetchTodosSuccess(todosData));
+      } catch (fetchError) {
+        dispatch(fetchTodosFailure('Error fetching todos'));
+      }
+    };
```

Comment:
This `useEffect` hook is responsible for fetching todos from the server. It is recommended to move such logic to Redux middleware (like redux-thunk or redux-saga) to keep the component only responsible for rendering and dispatching actions.